modules/post/linux/gather/enum_nagios_xi.rb

```diff
@@ -0,0 +1,326 @@
+  ##
+  # This module requires Metasploit: https://metasploit.com/download
+  # Current source: https://github.com/rapid7/metasploit-framework
+  ##
+
+  class MetasploitModule < Msf::Post
+    include Msf::Post::Linux::System
+    include Msf::Exploit::FileDropper
+
+    def initialize(info={})
+      super(update_info(info, {
+        'Name'           => 'Nagios XI Enumeration',
+        'Description'    => %q{
+          This module gathers saved credentials of NagiosXI monitored hosts and services.
+        },
+        'License'        => MSF_LICENSE,
+        'Author'         =>         [
+            'Cale Smith',   # @0xC413
+            'Benny Husted', # @BennyHusted
+            'Jared Arave'   # @iotennui
+          ],
+        'DisclosureDate'  => 'Aug 8 2017',
+        'Platform'       => 'linux',
+        'Arch'           => [ARCH_X86, ARCH_X64],
+        'Targets'        =>        [
+            ['Nagios XI 5.2.6 <= 5.4.12', upper_version: Gem::Version.new('5.4.12'), lower_version: Gem::Version.new('5.2.6')]
+        ],
+        'DefaultOptions' => { 'PrependFork' => true, 'WfsDelay' => 10 },
+        'SessionTypes'   => ['shell', 'meterpreter'],
+        'DefaultTarget'  => 0
```

Comment:
Most of these options (`DefaultTarget`, `DefaultOptions`, `Targets`, etc) are likely redundant in the `Post` context.